vllm/v1/core/sched/scheduler.py

```diff
@@ -402,6 +402,15 @@ def schedule(self) -> SchedulerOutput:
                             < num_new_tokens):
                         num_new_tokens = (
                             self.scheduler_config.long_prefill_token_threshold)
+
+                    # chunked prefill has to be enabled explicitly to allow
+                    # pooling requests to be chunked
+                    if not self.scheduler_config.chunked_prefill_enabled and \
+                        num_new_tokens > token_budget:
+                        self.waiting.popleft()
+                        skipped_waiting_requests.appendleft(request)
+                        continue
```

Comment:
I'm not sure this could potentially cause a starvation problem, but it seems to be ok at least for the pooling/embedding models